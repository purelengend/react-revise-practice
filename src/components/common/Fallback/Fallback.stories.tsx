import type { Meta, StoryObj } from "@storybook/react";
import { Center } from "@chakra-ui/react";

// Components
import Fallback from ".";

const meta: Meta<typeof Fallback> = {
  title: "Components/Fallback",
  component: Fallback,
  decorators: [
    (Story) => (
      <Center style={{ width: "100vh" }}>
        <Story />
      </Center>
    ),
  ],

  parameters: {
    controls: {
      expanded: true,
    },
  },
};

export default meta;

type Story = StoryObj<typeof Fallback>;

export const Primary: Story = {};
