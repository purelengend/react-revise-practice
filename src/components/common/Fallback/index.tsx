import { Text } from "@chakra-ui/react";

const Fallback = () => {
  return (
    <Text textAlign="center" fontSize="2xl" color="red">
      Something went wrong.
    </Text>
  );
};

export default Fallback;
