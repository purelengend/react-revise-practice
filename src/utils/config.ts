// Check import.meta.env is undefined in test environment
export const required = (key: string, defaultValue = undefined) => {
  if (import.meta.env) return import.meta.env[key];
  return defaultValue;
};
