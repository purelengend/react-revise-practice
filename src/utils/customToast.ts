import { TOAST_DEFAULT_OPTIONS } from "@/constants";

import { AlertStatus, UseToastOptions } from "@chakra-ui/react";
export const customToast = (
  title: string,
  description: string,
  status: AlertStatus,
): UseToastOptions => {
  if (status === "success") {
    return {
      ...TOAST_DEFAULT_OPTIONS.SUCCESS,
      title,
      description,
    };
  }

  return {
    ...TOAST_DEFAULT_OPTIONS.ERROR,
    title,
    description,
  };
};
