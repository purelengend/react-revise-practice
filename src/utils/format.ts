export const formatPhoneNumber = (phoneNumber: string) => {
  // Check if the phone number is valid (10 digits)
  if (!/^\d{10}$/.test(phoneNumber)) {
    throw new Error("Invalid phone number. Must be a 10-digit number.");
  }

  // Format the phone number
  const areaCode = phoneNumber.slice(0, 3);
  const middlePart = phoneNumber.slice(3, 6);
  const lastPart = phoneNumber.slice(6, 10);

  return `${areaCode}-${middlePart}-${lastPart}`;
};
