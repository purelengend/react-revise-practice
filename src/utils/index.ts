export * from "./test-utils";
export * from "./config";
export * from "./customToast";
