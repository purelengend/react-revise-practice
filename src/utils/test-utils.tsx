/* eslint-disable react-refresh/only-export-components */
import { RenderOptions, render } from "@testing-library/react";
import { ReactElement, ReactNode } from "react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { MemoryRouter } from "react-router-dom";

const queryClient = new QueryClient({});
export const AllTheProviders = (props: { children: ReactNode }) => {
  return (
    <MemoryRouter initialEntries={["/"]}>
      <QueryClientProvider client={queryClient}>
        {props.children}
      </QueryClientProvider>
    </MemoryRouter>
  );
};

export const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
