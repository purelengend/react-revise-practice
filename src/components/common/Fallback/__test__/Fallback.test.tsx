import { customRender } from "@/utils";

// Components
import Fallback from "..";

describe("Fallback component test case", () => {
  const setup = () => customRender(<Fallback />);

  it("should render correctly", () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
