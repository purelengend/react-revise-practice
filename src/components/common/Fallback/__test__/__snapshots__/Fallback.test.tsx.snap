// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fallback component test case should render correctly 1`] = `
<div>
  <p
    class="chakra-text css-1vhs2ha"
  >
    Fallback
  </p>
</div>
`;
